### DISCUSSION TITLE
Github markdown alerts support

### DISCUSSION THREAD

**Discussion by prescottprue** [withastro/starlight#1884] [Feature Requests]
### What version of `starlight` are you using?

0.22.4

### What is your idea?

As described [In Github's Docs for markdown](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) alerts (which are written in blockquote syntax) are rendered in a specific way which makes a clear callout to the reader.

Like so:
> [!TIP]
> Helpful advice for doing things better or more easily.

It would be great if Starlight supported matching this functionality in it's rendered output. It isn't required, but it would be a plus if there was an option (or a default) to have this actually rendered as [a Starlight Aside](https://starlight.astro.build/guides/authoring-content/#asides)

### Why is this feature necessary?

Markdown written on Github often contains these alerts now that they are supported, and currently in Starlight this is rendered as a blockquote which can easily have the information be lost (or at the very least, not follow it's original intention of being a callout to a reader).

### Do you have examples of this feature in other projects?

* [Github's Docs on Alerts](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts)
* Right in Github's rendering of this markdown:
> [!NOTE]
> Useful information that users should know, even when skimming content.

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for the suggestion @prescottprue!

I'm a bit hesitant about this one. GitHub's syntax is very nonstandard (and already changed once recently) and as far as I know, is not supported anywhere else. (Not even npm, which is part of the same company 😅).

I think I would wait and see if the GitHub syntax becomes more common before committing to supporting it. Especially as their variants don't align with our asides 1:1, so we'd need to think about how to handle their Important/Warning/Caution types.

It looks like https://github.com/hyoban/remark-github-alerts might work though? Could be added as a remark plugin in the Astro config to add this functionality for users who need it.

**Reply by prescottprue** [reply]
Totally fair on waiting for the API to stabalize before supporting - thanks for the clarity on decisioning. Also thanks for sharing remark-github-alerts, I'll give that a try

I can go ahead and close this discussion for now and we can wait to circle back

**Reply by delucis** [reply]
Thanks for the understanding! Let us know if `remark-github-alerts` works for you — could be helpful to others looking for this in the future.

**Comment by stereobooster** [comment]
Starlight supports different syntax for the same feature:

```md
:::note
Highlights information that users should take into account, even when skimming.
:::
```

**Reply by prescottprue** [reply]
@stereobooster Aware of that syntax - was hoping to discuss support for both. We have folks on our team moving over markdown which is rendered on Github currently and this requires a change in each place Github alerts are used.

**Comment by prescottprue** [comment]
Closing since [as @delucis called out](https://github.com/withastro/starlight/discussions/1884#discussioncomment-9463347) there is a preference to wait for the Github API to remain stable before supporting

**Comment by HiDeoo** [comment]
If ever needed, I just released a [new Starlight plugin](https://github.com/HiDeoo/starlight-github-alerts) to automatically render GitHub alerts as Starlight asides 📢
